core/api/src/app/accounts/get-default-wallet-for-username-phone.ts

```diff
@@ -0,0 +1,30 @@
+import { getWalletFromAccount } from "./get-wallet-from-account"
+
+import { CouldNotFindWalletFromUsernameAndCurrencyError } from "@/domain/errors"
+import { checkedToUsername } from "@/domain/accounts"
+import { checkedToPhoneNumber } from "@/domain/users"
+import { AccountsRepository, UsersRepository } from "@/services/mongoose"
+
+export const getDefaultWalletByUsernameOrPhone = async (
+  usernameOrPhone: Username | PhoneNumber,
+  walletCurrency?: WalletCurrency,
+): Promise<Wallet | ApplicationError> => {
+  const checkedUsername = checkedToUsername(usernameOrPhone)
+  if (!(checkedUsername instanceof Error)) {
+    const account = await AccountsRepository().findByUsername(checkedUsername)
+    if (account instanceof Error) return account
+    return getWalletFromAccount(account, walletCurrency)
+  }
+
+  const checkedPhoneNumber = checkedToPhoneNumber(usernameOrPhone)
+  if (checkedPhoneNumber instanceof Error)
+    return new CouldNotFindWalletFromUsernameAndCurrencyError(checkedPhoneNumber)
```

Comment:
```suggestion
    return new CouldNotFindWalletFromUsernameAndCurrencyError(usernameOrPhone)
```
please pass the original value, if we pass checkedPhoneNumber it will be another error